pkg/chunk/cache/instrumented.go

```diff
@@ -57,6 +57,10 @@ func (i *instrumentedCache) Store(ctx context.Context, keys []string, bufs [][]b
 	method := i.name + ".store"
 	instr.TimeRequestHistogram(ctx, method, requestDuration, func(ctx context.Context) error {
 		sp := ot.SpanFromContext(ctx)
+		if sp == nil {
+			sp, _ = ot.StartSpanFromContext(ctx, method)
```

Comment:
Why start a new span here?
Starting a span is expected of `TimeRequestHistogram` so I don't think these ones are necessary.